src/Microsoft.ServiceFabric.Actors/Runtime/ActorService.cs

```diff
@@ -69,6 +97,16 @@ public class ActorService : StatefulServiceBase, IActorService
             this.stateManagerFactory = stateManagerFactory ?? DefaultActorStateManagerFactory;
             this.actorManagerAdapter = new ActorManagerAdapter { ActorManager = new MockActorManager(this) };
             this.replicaRole = ReplicaRole.Unknown;
+
+            // Migration initialization
+            this.migrationOrchestrator = migrationOrchestrator;
+            if (this.migrationOrchestrator != null)
+            {
+                // Migration workflow is running
+                this.migrationOrchestrator.RegisterStateChangeCallback(stateChange => this.actorCallsAllowed = stateChange);
+                this.actorCallsAllowed = false;

```

Comment:
For Source service, actorCallsAllowed should be set to false when we enter DownTime phase. So here actorCallsAllowed should be set to false only for Target service

Reply:
The idea is that we start with false if migration attribute is present and from MigrationOrchestrator.StartMigrationAsync() we invoke the callback based on the current state of the stateprovider. Think of a scenario where migration is already completed and replica failed over. If we set it true by default, then replica will accept actor requests till the time migration orchestrator sets the flag to false.

Reply:
ok